src/services/TaskRepository.ts

```diff
@@ -0,0 +1,114 @@
+//import { Tasks } from "../models/ArrayModel";
+export class TaskRepo {
+    getTasks(completion:boolean, sortBy:string, sortOrder:string){
+        var queriedTasks:Task[]
+
+        switch (completion)
+        {
+            case true:
+                queriedTasks = Tasks.filter(Task => Task.completed == true);
+                break;
+            case false:
+                queriedTasks = Tasks.filter(Task => Task.completed == false);
+                break;
+            default:
+                queriedTasks = Tasks;
+        }
+
+        // Selects case based upon what query parameter was entered for sorting. 
+        switch (sortBy)
+        {
+            case 'createdAt':
+                
+                if (sortOrder == 'asc')
+                {
+                    queriedTasks.sort((a,b) => +new Date(a.createdAt) - +new Date(b.createdAt));
+                }
+                else if (sortOrder == 'desc')
+                {
+                    queriedTasks.sort((a,b) => +new Date(b.createdAt) - +new Date(a.createdAt));
+                }
+                break;
+
+            case 'dueDate':
+                if (sortOrder == 'asc')
+                {
+                    queriedTasks.sort((a,b) => +new Date(a.dueDate) - +new Date(b.dueDate));
+                }
+                else if (sortOrder == 'desc')
+                {
+                    queriedTasks.sort((a,b) => +new Date(b.dueDate) - +new Date(a.dueDate));
+                }
+                break;
+        }
+        return queriedTasks;
+    }
+
+    fetchTask(id:number){
+        return Tasks[Tasks.findIndex(Task => Task.id == id)]
+    }
+
+    createTask(desc:string, due:Date, completed: boolean){
+        var newTask = new Task((Tasks.length + 1), desc, new Date, due, completed)
```

Comment:
Relying on the length of the array to generate IDs is not viable because the length of the array is not necessarily correlated with the latest element in the list and this can lead to unintended behavior or crashes.

It would be better to find the highest `id` in array and then increment from that rather than the length of the store.